### ISSUE TITLE
RDKit Deprecation Warnings for MorganFingerprint implementations

### ISSUE THREAD

**Issue by RiesBen** [EBjerrum/scikit-mol#41]
Hi, 
not sure if that was already observed somewhere else, but I just recently stumbled over these warning:
```
[10:22:21] DEPRECATION WARNING: please use MorganGenerator
[10:22:21] DEPRECATION WARNING: please use MorganGenerator
[10:22:21] DEPRECATION WARNING: please use MorganGenerator
[10:22:21] DEPRECATION WARNING: please use MorganGenerator
[10:22:21] DEPRECATION WARNING: please use MorganGenerator
[10:22:21] DEPRECATION WARNING: please use MorganGenerator
[10:22:21] DEPRECATION WARNING: please use MorganGenerator
[10:22:21] DEPRECATION WARNING: please use MorganGenerator
```

I assume there is an RDKit change upcoming?

was using from conda-forge:
```
  Name      Version    Build            Channel    
─────────────────────────────────────────────────────
  librdkit  2024.03.5  h79cfef2_2       conda-forge
  rdkit     2024.03.5  py312h7b4b7d0_2  conda-forge
```
and via PyPi:

```
scikit_mol                    0.3.1
```

**Comment by EBjerrum** [comment]
Thanks for the heads-up. didn't see this warning.

**Comment by EBjerrum** [comment]
There'll be a redesigned base class needed and refactoring of most fingerprints . On the other hand, it will allow a more consistent implementation and opens up for some new features. A good overview of the options are in Gregs blogpost here:
[https://greglandrum.github.io/rdkit-blog/posts/2023-01-18-fingerprint-generator-tutorial.html](https://greglandrum.github.io/rdkit-blog/posts/2023-01-18-fingerprint-generator-tutorial.html)

**Comment by RiesBen** [comment]
Hackathon TIME ! :)

**Comment by EBjerrum** [comment]
Solved with #54 . Thanks for the effort!
